components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -3731,4 +3801,16 @@ bool RewardsServiceImpl::OnlyAnonWallet() {
   return false;
 }
 
+void RewardsServiceImpl::RecordBackendP3AStats() const {
+  // TODO(iefremov): It is not safe to use Unretained here, as well as
```

Comment:
it is actually safe to use `publisher_info_backend_` the way we are using it in the other methods (anonymous function) because `publisher_info_backend_` is destroyed by the task runner when it shuts down `file_task_runner_->DeleteSoon(FROM_HERE, publisher_info_backend_.release());`
so there is no way anything in the task runner can ever be called against a null `publisher_info_backend_`. I think this should be ok for the same reason, but I'd prefer to keep them consistent for now and use the same pattern because that has been running for a long time and without crashes afaik.

Reply:
I think the `null` check in the anonymous method is also not needed and wouldn't work correctly anyway because it already passed in the original pointer address, but it doesn't hurt anything

Reply:
I'd say we should refactor this pattern asap, because it uses more code and breaks CC (free functions are scattered across the code). But I've adjusted my piece to this style